### ISSUE TITLE
Multi-project setup confusion after version 0.7.4 (and still in 0.7.5)

### ISSUE THREAD

**Issue by JWvanV** [Kotlin/kotlinx-kover#522]
I have a (pretty big, but consistent) multi-project setup, that I needed help with setting up for the previous versions.
Now that things have changed again since version 0.7.4, and the documentation has not become more clear to me, I'd like to request help once again. I'm going to call the Android studio project the 'project' and each sub-project 'modules'.

Project structure
```
:app (Android)

:core:data (Android)
:core:domain (JVM)
:core:presentation (Android)

:featurex:data (Android)
:featurex:domain (JVM)
:featurex:presentation (Android)

... Other features...
```

The `app`, `data`, and `presentation` modules have different flavors, but this is pretty consistent:
- Each `presentation` module has a `release` build flavor that I'd like to use for Kover tests coverage.
- Each `data` module has a `tstRelease` build flavor that I'd like to use for Kover test coverage.
- The `app` module also has a `tstRelease` build flavor.

Current configuration:
- Each module (Android and JVM) has the Kover plugin applied
- As suggested last time: the `:app` module contains the following

```
dependencies {
    forEachModuleLayer { // (All above-mentioned project modules)
        implementation(project(it))
        kover(project(it))
    }
   ...
}

koverReport {
    // filters for all report types of all build variants
    filters {
        ...
    }
    defaults {
        mergeWith("tstRelease")
        ...
    }
}
```

The pipeline runs this as:
`./gradlew app:koverXmlReport app:koverHtmlReport`
which generates the expected single coverage output for the entire project.

Now, after updating to Kover 0.7.5, I get the following error:
```
> Kover android variant 'tstRelease' was not matched with any variant from dependency ':core:domain' of project ':app'. Check that the Kover plugin is applied in the ':core:domain' project and there is a variant compatible with 'tstRelease' in it.
```

The Kover plugin has been applied to `:core:domain`, but adding this to the `core/domain/build.gradle.kts` also doesn't work:
```
koverReport {
    defaults{
        mergeWith("tstRelease") // Or "release" for that matter
    }
}
```
It gives me the error:
```
Build variant 'tstRelease' not found in project ':core:domain' - impossible to merge default reports with its measurements.
Available variations: []
```

In short, I'm stuck and do now know how to proceed from here. Could you guys point me in the right direction?

**Comment by shanshin** [comment]
Hi,
[this comment](https://github.com/Kotlin/kotlinx-kover/issues/509#issuecomment-1830635724) provides recommendations for creating a merged report for Android + JVM projects.

The first thing you need to change is to move the code with `kover(project(it))` dependencies to a project where no Android plugin is used. Most often, the most appropriate place is the root project.
In this case, you can add a dependency by iterating through `if allowed` (if allowed).

Then you need to add it to all `presentation` projects 
```
koverReport {
    defaults{
        mergeWith("release")
    }
}
```
also in all `data` and `app` projects
```
koverReport {
    defaults{
        mergeWith("tstRelease")
    }
}
```


Specify reports filters
```
koverReport {
    // filters for all report types of all build variants
    filters {
        ...
    }
}
```
only in the project in which you wrote `kover(project(it))` (preferably in the root project)


So, if the combined report is located in the root project, then you can generate it with the `:koverHtmlReport` command

**Comment by JWvanV** [comment]
This worked :D
Thanks for the help!

**Comment by cj1098** [comment]
Hey @JWvanV just curious what your `forEachModuleLayer {}` dsl is? Is it a custom gradle plugin?

**Comment by JWvanV** [comment]
Oh, just a function I wrote inside of BuildSrc.
Because of the project structure (see OP) and having about 9 features, all with 3 sub-modules (layers), I didn't want to have to write 27 'implementation' lines. (doubled if I would have to do both an implementation line and a kover line)
So that function just generates a module string for each of the 27 module-layers (combines a list of 9 feature names and a list of the 3 layer names) and then runs a foreach with an optional block.
So no magic DSL here, just a helper function for my project structure :)
